.travis.yml

```diff
@@ -8,26 +8,32 @@ php:
   - 7.0
   - hhvm
 
-matrix:
-  allow_failures:
-    - php: 7.0
-    - php: hhvm
+sudo: false
 
 cache:
   directories:
     - $HOME/.composer/cache
+    - vendor
```

Comment:
> The cache’s purpose is to make installing language-specific dependencies easy and fast

Right.

> but our concern is to test the codebase against many changing vendors dependencies.

Wrong, or I don't get it. Vendor caching is here just to avoid re-installation of **already up to date** dependencies according to `composer.json` file.

Caching vendor or not, the behavior will be the same: All dependencies will be updated if necessary.

The advantage is that only needed package are changed, that make going Travis faster.

> This also has an unresolved side effects over the `Makefile` tasks.

Which one? And BTW, Makefile is not used on Travis AFAIK.

> In addition, as a maintainer, I don't want to be asked to manually clear the Travis cache every time a PR encounter a BC break issue and the PR author wants to digg into a reliable travis job trace.

Cache clearing would not be needed. I ask for it here because of `dir` bin issue. It's a special case.


I hope I'm enough clear on my explanation. Don't hesitate to ask me if not.